### PR TITLE
Slash Management

### LINKED ISSUE #768: Utility - Add - slash management
The code has several implementations of slash management for input configs such as what was built in #699 and elsewhere for removing duplicate slashes, trailing slashes, etc.

Create utility that will manage paths/uris/url slash logic. Function should at least: 
- remove duplicate slashes 
- remove trailing slashes
- handle file uris appropriately such as abfss:/ -> abfss:// etc.

### PR DESCRIPTION
**Pull request by sriram251-code**
closes #768

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 1]
```diff
@@ -881,5 +881,32 @@ object Helpers extends SparkSessionWrapper {
 
   }
 
+  def sanitizeURL(url:String):String={
+    val inputUrl = url.trim
+    removeDuplicateSlashes(removeTrailingSlashes(inputUrl))
+  }
+
+  def removeDuplicateSlashes(url: String): String = {
```
Comment (souravbaner-da) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:889]: Add Documentation

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 2]
```diff
@@ -881,5 +881,32 @@ object Helpers extends SparkSessionWrapper {
 
   }
 
+  def sanitizeURL(url:String):String={
```
Comment (souravbaner-da) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:884]: Add Documentation

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 3]
```diff
@@ -881,5 +881,32 @@ object Helpers extends SparkSessionWrapper {
...
+        stringURL.contains("http:/") ||
+        stringURL.contains("https:/")) true else false
+    if (makeFirstSlashDoubleSlash) {
+      stringURL.replaceFirst("/", "//")
+    } else {
+      stringURL
+    }
+  }
 
+  def removeTrailingSlashes(url: String): String = {
```
Comment (souravbaner-da) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:905]: Add Documentation

**Review by souravbaner-da — CHANGES_REQUESTED** [review]
Commend added for the required changes

**Review by souravbaner-da — APPROVED** [review]

**Review by GeekSheikh — CHANGES_REQUESTED** [review]
So you built the function to manage slashes -- GREAT -- but it's implemented in exactly 0 places. Please identify all the location where we manually manage slashes and replace with this function appropriately.

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala [hunk 4]
```diff
@@ -860,11 +860,11 @@ trait BronzeTransforms extends SparkSessionWrapper {
...
     try{
     val mountMap = getMountPointMapping(apiEnv) //Getting the mount info from api and cleaning the data
-      .withColumn("mount_point", when('mount_point.endsWith("/"), 'mount_point.substr(lit(0), length('mount_point) - 1)).otherwise('mount_point))
-      .withColumn("source", when('source.endsWith("/"), 'source.substr(lit(0), length('source) - 1)).otherwise('source))
+      .withColumn("mount_point", removeTrailingSlashes('mount_p))
+      .withColumn("source",  removeTrailingSlashes('source))
       .filter(col("mount_point") =!= "/")
     //Cleaning the data for cluster log path
-    val formattedInputDf = inputDataframe.withColumn("cluster_log_conf", when('cluster_log_conf.endsWith("/"), 'cluster_log_conf.substr(lit(0), length('cluster_log_conf) - 1)).otherwise('cluster_log_conf))
+    val formattedInputDf = inputDataframe.withColumn("cluster_log_conf",  removeTrailingSlashes('cluster_log_conf))
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:867]: Love how much simpler this is now.

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala [hunk 5]
```diff
@@ -860,11 +860,11 @@ trait BronzeTransforms extends SparkSessionWrapper {
   private[overwatch] def getAllEventLogPrefix(inputDataframe: DataFrame, apiEnv: ApiEnv): DataFrame = {
     try{
     val mountMap = getMountPointMapping(apiEnv) //Getting the mount info from api and cleaning the data
-      .withColumn("mount_point", when('mount_point.endsWith("/"), 'mount_point.substr(lit(0), length('mount_point) - 1)).otherwise('mount_point))
-      .withColumn("source", when('source.endsWith("/"), 'source.substr(lit(0), length('source) - 1)).otherwise('source))
+      .withColumn("mount_point", removeTrailingSlashes('mount_p))
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:863]: column `mount_p` doesn't exist this will break the code.

**Review by GeekSheikh — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 5 above)
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala]: yes you are 100% right. Mistake by my end while doing final push. By mistake the characters were removed my bad. But before doing the final push i have done a deployment and it worked fine.

**Review by GeekSheikh — APPROVED** [review]

**Pull request merged**